bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableDataSettingsFactory.java

```diff
@@ -0,0 +1,275 @@
+import com.google.cloud.bigtable.config.BigtableOptions;
+import com.google.cloud.bigtable.config.BulkOptions;
+import com.google.cloud.bigtable.config.CredentialFactory;
+import com.google.cloud.bigtable.config.CredentialOptions;
+import com.google.cloud.bigtable.config.Logger;
+import com.google.cloud.bigtable.config.RetryOptions;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings.Builder;
+import com.google.cloud.bigtable.data.v2.models.InstanceName;
+import com.google.cloud.bigtable.data.v2.stub.BigtableStubSettings;
+
+import io.grpc.ManagedChannelBuilder;
+
+/**
+ * Static methods to convert an instance of {@link Configuration} or {@link BigtableOptions} to a
+ * {@link BigtableDataSettings} instance.
+ */
+public class BigtableDataSettingsFactory {
+  /** Constant <code>LOG</code> */
+  private static final Logger LOG = new Logger(BigtableDataSettingsFactory.class);
+
+  /**
+   * To create an instance of {@link BigtableDataSettings} from {@link BigtableOptions}.
+   *
+   * @param options a {@link BigtableOptions} object.
+   * @return a {@link BigtableDataSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableDataSettings fromBigtableOptions(final BigtableOptions options)
+      throws IOException, GeneralSecurityException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+    checkState(options.getRetryOptions().enableRetries(), "Disabling retries is not currently supported.");
+
+    BigtableDataSettings.Builder builder = BigtableDataSettings.newBuilder();
+
+    InstanceName instanceName = InstanceName.newBuilder().setProject(options.getProjectId())
+        .setInstance(options.getInstanceId()).build();
+    builder.setInstanceName(instanceName);
+    builder.setAppProfileId(options.getAppProfileId());
+
+    builder.setEndpoint(options.getDataHost() + ":" + options.getPort());
+
+    buildCredentialProviderSettings(builder, options.getCredentialOptions());
+
+    buildBulkMutationsSettings(builder, options);
+
+    buildCheckAndMutateRowSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadModifyWriteSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadRowsSettings(builder, options);
+
+    buildMutateRowSettings(builder, options);
+
+    buildSampleRowKeysSettings(builder, options);
+
+    // TODO: implementation for channelCount or channelPerCPU
+    ManagedChannelBuilder channelBuilder = ManagedChannelBuilder
+        .forAddress(options.getDataHost(), options.getPort())
+        .userAgent(options.getUserAgent());
+
+    if (options.usePlaintextNegotiation()) {
+      channelBuilder.usePlaintext();
+    }
+
+    builder.setTransportChannelProvider(
+      FixedTransportChannelProvider.create(GrpcTransportChannel.create(channelBuilder.build())));
+
```

Comment:
@ajaaym I would need a little guidance for netty server setup for UserAgent JUnit test case.

***
**This PR is to track progress & review changes**